DEV: dry up the Do Not Disturb widget

AndrewPrigorshnev opened the pull request.

nbianca commented on app/assets/javascripts/discourse/app/widgets/do-not-disturb.js:
```diff
@@ -11,21 +11,10 @@ export default createWidget("do-not-disturb", {
-      ];
-    }
+    const isOn = this.currentUser.isInDoNotDisturb();
+    const button = this._menuButton(isOn);
+
+    return [button];
```
Just a personal preference, but this function looks longer than it should be.

```suggestion
    return [this._menuButton(isOn)];
```

nbianca commented on app/assets/javascripts/discourse/app/widgets/do-not-disturb.js:
```diff
@@ -11,21 +11,10 @@ export default createWidget("do-not-disturb", {
-      return [
-        h("button.btn-flat.do-not-disturb-inner-container", [
-          iconNode("toggle-off"),
-          h("span.do-not-disturb-label", I18n.t("pause_notifications.label")),
```
I do not see this string used anymore if the "do not disturb" mode is not enabled. Can you confirm it is still in use?

AndrewPrigorshnev replied:
We use it in a couple of places: below in the `_label` function and also in the `profile-tab-content` component.

jjaffeux commented:
@AndrewPrigorshnev I think we should try to merge this Monday

AndrewPrigorshnev commented:
Test failures are unrelated, we're having them on other PRs as well currently. @jjaffeux or @nbianca could you please approve it, I've addressed review comments.

AndrewPrigorshnev commented:
Oh, this actually was already approved, forgot about it.
